DataFormats/PatCandidates/interface/TriggerObjectStandAlone.h

```diff
@@ -147,10 +156,33 @@ namespace pat {
       /// Calls 'hasCollection(...)' (method override)
       virtual bool coll( const std::string & collName ) const { return hasCollection( collName ); };
 
+      ///set the HLT process name used to fill the object
+      void setProcessName(std::string const& processName){processName_=processName;}
+      ///set the psetid of the trigger process
+      void setPSetID(const edm::ParameterSetID &psetId){psetId_=psetId;}
+      
+      const edm::ParameterSetID &psetID() const {return psetId_;}
+
+
+      void packFilterLabels() {packFilterLabels(*allLabels(psetId_));}
       ///  pack trigger names into indices 
       void packPathNames(const edm::TriggerNames &names) ;
       ///  unpack trigger names into indices 
       void unpackPathNames(const edm::TriggerNames &names) ;
+      ///  pack filter labels into indices; note that the labels must be sorted!
+      void packFilterLabels(const std::vector<std::string> &labels) const ;
+      ///  unpack filter labels into indices
+      void unpackFilterLabels(const std::vector<std::string> &labels) const ;
+
+      void unpackFilterLabels() const ;
+
+
+      /// reduce the precision on the 4-vector
+      void packP4();
+
+//      std::vector<std::string> const* allLabels(edm::TriggerResults const& triggerResults,
+  //                                              edm::ProcessHistory const &history,std::string const& processName="HLT") ;
+      std::vector<std::string> const* allLabels(edm::ParameterSetID const& psetid) const ;
```

Comment:
yes... I'm was sure I made it static... but I clearly did not... but do we really want it to live here? or should we have it in Event (same as for triggerNames) ?
This will also ensure that no one tries to access the labels info from TTree::Draw and rather need a valid fwlite::Event (or edm::Event) object in hand